quinn-proto/src/endpoint.rs

```diff
@@ -409,9 +404,10 @@ impl Endpoint {
                 ).saturating_sub(RESET_TOKEN_SIZE),
             );
             buf.reserve_exact(header_len + padding + RESET_TOKEN_SIZE);
+            let number = self.ctx.rng.gen::<u32>() & 0x3fffffff | 0x4000;
```

Comment:
I've named the former one. I'm not sure the latter one is needed, what do you think? Is it a requirement that packet numbers are encoded in their shortest possible encoding?

Reply:
On review, [the spec](https://quicwg.org/base-drafts/draft-ietf-quic-transport.html#rfc.section.10.4) now calls for the first byte of the packet to be followed by "at least 20 bytes of random or unpredictable values" (including the random padding bytes such as we already generate) rather than a well-formed header. That seems like the simpler way to proceed.